### PR TITLE
Clickhouse Sessions + Frontend Sessions updates

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Frontend
  - Convert Sessions to TypeScript
  - Add selected day to URL to make back/forward work
  - Add back/forward buttons to the date selection
  - Simplify date parameter sent to the backend (might have a timezone issue?)
  - Refactor the logic quite a bit
  - Fix bug with order and timestamps in pagination of events

- Backend
  - Add limit/offset support for clickhouse sessions query
  - Add support for fetching events inside the sessions query
  - Refactor how the Postgres sessions query does pagination
  - Add Elements into the clickhouse response (might do async in frontend? or join in query? not sure)
  - Add user properties to the clickhouse response

## Checklist

- [ ] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [ ] Backend tests (if this PR affects the backend)
- [ ] Cypress E2E tests (if this PR affects the front and/or backend)

### CODE REVIEW

**Comment by mariusandra** [comment]
Elements are now in the query and almost everything works. There's still a bug with the "order" column if the events list is paginated... and I need to get user properties into [the query](https://github.com/PostHog/posthog/blob/cd9d2423130317deee655cdd6abef3e2c2e1fde9/ee/clickhouse/queries/clickhouse_sessions.py#L20) somehow.

@EDsCODE @fuziontech what's the best way to get user properties now? What tables/materialized_views/etc should I get this from?

image

**Comment by mariusandra** [comment]
The last todo items are fixed now. Ready for a review!

File: posthog/api/insight.py [hunk 1]
```diff
@@ -152,12 +153,14 @@ def session(self, request: request.Request, *args: Any, **kwargs: Any) -> Respon
...
+        limit = SESSIONS_LIST_DEFAULT_LIMIT + 1
+        result: Dict[str, Any] = {"result": sessions.Sessions().run(filter=filter, team=team, limit=limit)}
 
         # add pagination
         if filter.session_type is None:
-            offset = filter.offset + 50
-            if len(result["result"]) > 49:
+            offset = filter.offset + limit - 1
+            if len(result["result"]) > SESSIONS_LIST_DEFAULT_LIMIT:
+                result["result"].pop()
```
Comment (EDsCODE) [inline posthog/api/insight.py:163]: fun!

**Review by EDsCODE — COMMENTED** [review]
looks good! 

would be good to add a proper test for pagination session list now

**Comment by mariusandra** [comment]
Will do, though only tomorrow morning. Feel free to merge it in until then if you need the stuff in master already

**Comment by timgl** [comment]
Also failing mypy

**Comment by mariusandra** [comment]
I added a test to test the session pagination... and I enabled almost all of the insight API tests in clickhouse. 

The ones failing were funnels and retention. So I commented them out for now (only when testing EE) and added a `TODO` to fix this later.

image

However, locally, I can't run the API tests unless I comment out these lines:

image

The error `e` says `cannot import name 'ClickhouseActions' from 'ee.clickhouse.views'` and all tests that make an API call fail with the error `Endpoint not found`.

Any idea why this could be happening?

**Comment by EDsCODE** [comment]
Doesthis happen when you specifically target a test to run like ‘ee.clickhouse.views.test....’ but not when you run ‘ee.clickhouse’?

There’s some sort of circular dependency happening I believe with the barrel file for views. @timgl fixed it in the work he’s doing.

**Comment by mariusandra** [comment]
You're right. If I run all the tests, everything passes. I guess this is safe to merge...

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**